sllurp/llrp.py

```diff
@@ -1737,18 +1738,17 @@ def on_lost_connection(self):
             self._on_disconnected()
             return True
 
-        while max_retry:
-            max_retry -= 1
+        while remaining_attempts > 0 or remaining_attempts == -1:
+            if remaining_attempts == 0:
+                logger.info('Too many retries. Giving up...')
+                break
             try:
                 self._connect_socket()
                 return False
             except:
                 logger.warning('Reconnection attempt failed.')
-
-            if max_retry <= 0:
-                logger.info('Too many retries. Giving up...')
-                break
-
+            if(remaining_attempts > 0):
+                remaining_attempts -= 1
```

Comment:
In fact, a little bit cleaner, what I think you should do:
- keep
`while max_retry:` 
because it will be true for all cases except "0".
...

```
if remaining_attempts == -1:
  continue
remaining_attemps -= 1
if remaining_attemps <= 0:
   ... logger.info...
   break

```

Reply:
Btw, as it is not C, when you have simple if statements without any order doubt, don't use parenthesis.
For example:
`if(remaining_attempts > 0):`
=> `if remaining_attempts > 0:`

Reply:
would something like this makes more sense. No quite sure why we would need both max_retry and remaining_attempts

```
        while True:
            if remaining_attempts == 0:
                logger.info('Too many retries. Giving up...')
                break
            else:
                try:
                    self._connect_socket()
                    return False
                except:
                    logger.warning('Reconnection attempt failed.')
                if remaining_attempts > 0:
                    remaining_attempts -= 1
                logger.info('Next connection attempt in %ds', retry_delay)
                user_disconnected = self.disconnect_requested.wait(retry_delay)
                if user_disconnected:
                    # Disconnection was requested by user
                    break
```

Reply:
I'm sorry, i writed a little bit too fast, but I was meaning:
Keep:
`while remaining_attemps:`

As the value will only be "False" when the value is == 0.
For the rest of the change, the goal is to avoid as much as possible be in "if" sub-blocks.
Also, you can consider that each time you enter inside the while, you are in a valid attempt >0 or infinite, so you can do the attempt and check for the retry maximum at the end of the block.